var/spack/repos/builtin/packages/cabana/package.py

```diff
@@ -63,34 +64,49 @@ class Cabana(CMakePackage, CudaPackage, ROCmPackage):
                 _kk_spec = "kokkos{0}+{1}".format(_kk_version, _backend)
             depends_on(_kk_spec, when="@{0}+{1}".format(_version, _backend))
 
+    # Propagate cuda architectures down to Kokkos and optional submodules
     for arch in CudaPackage.cuda_arch_values:
         cuda_dep = "+cuda cuda_arch={0}".format(arch)
         depends_on("kokkos {0}".format(cuda_dep), when=cuda_dep)
+        depends_on("heffte {0}".format(cuda_dep), when="+heffte {0}".format(cuda_dep))
+        depends_on("arborx {0}".format(cuda_dep), when="+arborx {0}".format(cuda_dep))
+        depends_on("hypre {0}".format(cuda_dep), when="+hypre {0}".format(cuda_dep))
 
     for arch in ROCmPackage.amdgpu_targets:
         rocm_dep = "+rocm amdgpu_target={0}".format(arch)
         depends_on("kokkos {0}".format(rocm_dep), when=rocm_dep)
+        depends_on("heffte {0}".format(rocm_dep), when="+heffte {0}".format(rocm_dep))
+        depends_on("arborx {0}".format(rocm_dep), when="+arborx {0}".format(rocm_dep))
+        depends_on("hypre {0}".format(rocm_dep), when="+hypre {0}".format(rocm_dep))
 
     conflicts("+cuda", when="cuda_arch=none")
+    conflicts("+rocm", when="amdgpu_target=none")
+
     depends_on("kokkos+cuda_lambda", when="+cuda")
 
+    # Dependencies for subpackages
     depends_on("arborx", when="@0.3.0:+arborx")
     depends_on("hypre-cmake@2.22.0:", when="@0.4.0:+hypre")
     depends_on("hypre-cmake@2.22.1:", when="@0.5.0:+hypre")
-    # Previous heFFTe pinned at 2.x.0 because its cmakefiles can't roll forward
-    # compatibilty to later minor versions.
     depends_on("heffte@2.0.0", when="@0.4.0+heffte")
-    depends_on("heffte@2.1.0", when="@0.5.0:+heffte")
+    depends_on("heffte@2.1.0", when="@0.5.0+heffte")
     depends_on("heffte@2.3.0:", when="@0.6.0:+heffte")
     depends_on("silo", when="@0.5.0:+silo")
     depends_on("hdf5", when="@0.6.0:+hdf5")
     depends_on("mpi", when="+mpi")
 
+    # Cabana automatically builds HDF5 support with newer cmake versions
+    conflicts("~hdf5", when="^cmake@:3.26")
```

Comment:
I believe this was fixed in #40168 (adding the `DISABLE_FIND_PACKAGE_*` below) - this would require hdf5 with new cmake, correct?

Reply:
This does. The reason why is that the current Cabana Cmake unconditionally adds an HDF5 dependency if you're on CMake 3.26 or newer:

`if(CMAKE_VERSION VERSION_LESS 3.26)
  option(Cabana_REQUIRE_HDF5 "Require Cabana to build with HDF5 support" OFF)
  if(Cabana_REQUIRE_HDF5)
    # Workaround for using HDF5 C-bindings
    enable_language(C)
    find_package( HDF5 REQUIRED COMPONENTS C )
  endif()
  set(Cabana_ENABLE_HDF5 ${Cabana_REQUIRE_HDF5})
else()
  Cabana_add_dependency( PACKAGE HDF5 COMPONENTS C )
endif()`

Reply:
That `Cabana_add_dependency` should still respect `DISABLE_FIND_PACKAGE_HDF5` (but I assume you added this due to still having problems) ECP-CoPA/Cabana#699 does fix the underlying issue, but that will of course not fix 0.6

Reply:
Yeah sorry, you're right

Reply:
Changed this to conflict only on "@0.6.0 ^cmake@3.26:".